File: lib/core/constants/backend_config.dart
/// Backend URL resolution — supports .env, --dart-define, and production fallback.
library;

const String kProductionBackendUrl = 'https://weathergpt-backend.vercel.app';
const String kEmulatorBackendUrl = 'http://10.0.2.2:8888';

String resolveBackendUrl({String? dartDefineUrl, String? dotenvUrl}) {
  String? pick(String? raw) {
    if (raw == null) return null;
    var v = raw.trim();
    if (v.isEmpty) return null;
    if ((v.startsWith('"') && v.endsWith('"')) || (v.startsWith("'") && v.endsWith("'"))) {
      v = v.substring(1, v.length - 1).trim();
    }
    v = v.replaceAll(RegExp(r'/+$'), '');
    return v.isEmpty ? null : v;
  }
  return pick(dartDefineUrl) ?? pick(dotenvUrl) ?? kProductionBackendUrl;
}

File: src/core/config/backendConfig.ts
/// Backend URL resolution — port of `lib/core/constants/backend_config.dart`.

export const PRODUCTION_BACKEND_URL = "https://weathergpt-backend.vercel.app";

function pick(raw: string | null | undefined): string | null {
  if (raw === null || raw === undefined) return null;
  let v = raw.trim();
  if (v === "") return null;
  if ((v.startsWith('"') && v.endsWith('"')) || (v.startsWith("'") && v.endsWith("'"))) {
    v = v.slice(1, -1).trim();
  }
  v = v.replace(/\/+$/, "");
  return v === "" ? null : v;
}

/// `defineUrl` mirrors the Dart `--dart-define` slot: EXPO_PUBLIC_BACKEND_URL
/// in this port. `dotenvUrl` is the bundled .env value.
export function resolveBackendUrl(opts?: { defineUrl?: string | null; dotenvUrl?: string | null }): string {
  const defineUrl = opts?.defineUrl ?? null;
  const dotenvUrl = opts?.dotenvUrl ?? null;
  return pick(defineUrl) ?? pick(dotenvUrl) ?? PRODUCTION_BACKEND_URL;
}
